### ISSUE TITLE
XML Session Read & Write Failures

### ISSUE THREAD

**Issue by TEKgeist** [ictinnovations/ictfax#45]
I have followed all of the guides, and I have also used the installation script.  I have checked all of forums, GITHUB help, and every available help out there.  I overcome the bug in the ICTCORE.XML code within the ICTCORE FreeSwitch Dialplan.  FreeSwitch service now loads correctly without crashing on this new version.

Also, I was able to fix the MYSQL connections issues to localhost, and it's not throwing any connection errors.  I have reinstalled about 7 times using 3 different droplet rebuilds, and I still can't get past this bug.  I'm running a DigitalOcean Centos 7.6 droplet using the installation guide with ICTFAX 4.3.0.3 installed via YUM using the installation guide.

Within my ICTCORE Log, any time I try to update any Providers or any other data within the ICTFax Web Interface, I am getting these XML and Session Write Failure errors.  But, the Provider XML files are not writing at all to the _**/usr/ictcore/etc/freeswitch/sip_profiles/provider**_ folder.

_[2020-07-02 18:15:24] [967] [WARNING] Session read found zero rows
[2020-07-02 18:15:24] [963] [WARNING] Session read found zero rows
[2020-07-02 18:15:25] [963] [WARNING] Session write failed
[2020-07-02 18:15:35] [969] [WARNING] DOMDocument::loadXML(): xmlParseEntityRef: no name in Entity, line: 4 in /usr/ictcore/core/Gateway/Freeswitch.php on line 200
[2020-07-02 18:15:35] [969] [INFO] Freeswitch saving config for type: sip, name: provider_1
[2020-07-02 18:15:35] [969] [WARNING] Session write failed_

Freeswitch is not making any registration attempts, but I do see activity on FS_CLI as soon as I press SAVE in the web interface.  It's recognizing the XML file is being re-written, but the XML is lacking any data.  Within the provider XML, it's blank.  It looks like this.

_< ?xml version="1.0"? >_

**_What could cause this?  Web permissions?  Are there any tests I can run to see what's failing?  Should I be installing an older version via GITHUB?_**

Please let me know ASAP.  Thanks for your help.

**Comment by fizaict** [comment]
Can you please tell your ICTCore Version?

**Comment by TEKgeist** [comment]
I manually created the I ran <yum list ictcore>.  Is this what you were looking for?  

**Installed Packages
ictcore.noarch             1.0.3.17deead-main.el7            @ict**

So, I still have the issue.  But, I created some debugging using your Corelog:log commands.  I added the following line to your existing Core/Gateway/Freeswitch.php page for debugging purposes.  This gave me the output needed to manually copy and paste into the blank XML file.  Then, I copied and pasted that into the /usr/ictcore/etc/freeswitch/sip_profiles/provider/

 $doc = new DOMDocument();
    $doc->formatOutput = true;
    $doc->loadXML($data);
    **Corelog::log("Type = '$type' \n Name = '$name' \n Data = \n'$data'", Corelog::INFO);**
    Corelog::log("Freeswitch saving config for type: $type, name: $name", Corelog::INFO);
    $config_file = $this->config_filename($type, $name);
    $doc->save($config_file);

By doing this, I started getting some activity in Freeswitch.  I still can't get the incoming calls to connect properly and the outgoing are also failing.  I think the other dialplan XMLs are also failing to create which means I may have to debug and manually build those also.
